modules/base/include/modules/base/algorithm/camerautils.h

```diff
@@ -0,0 +1,209 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <modules/base/basemoduledefine.h>
+#include <inviwo/core/properties/cameraproperty.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/compositeproperty.h>
+#include <inviwo/core/properties/buttonproperty.h>
+#include <inviwo/core/ports/volumeport.h>
+#include <inviwo/core/ports/meshport.h>
+#include <inviwo/core/common/inviwo.h>
+
+#include <modules/base/algorithm/mesh/axisalignedboundingbox.h>
+
+namespace inviwo {
+
+namespace camerautil {
+
+enum class Side { XNegative, XPositive, YNegative, YPositive, ZNegative, ZPositive };
+
+/**
+ * Setup the camera parameters such that the whole boundingBox spaned by basis and offset
+ * will be inside the view frustum.
+ *
+ * @param cam       the Camera to update
+ * @param boundingBox the basis and offset of the bounding box that will fit inside the new
+ * view frustum
+ * @param side       this side of the bounding box will be facing the camera afterward
+ * @param fitRatio   determines the spacing between volume and the boundaries of the view frustum. A
+ *                   fit ratio of 1 means a perfect fit, no space between frustum and volume. The
+ * aspect ratio is taken into account.
+ * @param setNearFar the camera's new/far clip ranges are updated if true @see setCameraNearFar
+ * @param setLookRanges   the camera's look-to/look-from ranges are updated if true @see
+ * setCameraLookRanges
+ */
+IVW_MODULE_BASE_API void setCameraView(CameraProperty &cam, const mat4 &boundingBox, Side side,
+                                       float fitRatio = 1.05f, bool setNearFar = false,
+                                       bool setLookRanges = false);
```

Comment:
How about instead the of booleans:
Function overloads with the corresponding parameters given, e.g., 
fitObjectIntoView(cam, boundingBox, side, fitRatio) (equivalent to setNear, setLook = false)
fitObjectIntoView(cam, boundingBox, side, fitRatio, maxZoomOutFactor, nearFarRatio) (equivalent to setNear, setLook = true)

Reply:
Took the liberty of suggesting a function name change that I think is more expressive of what the function does